### PR TITLE
feat: remove defaulting to semver versioning

### LINKED ISSUE #10320: Remove regex manager versioning default
**What would you like Renovate to be able to do?**

Rescuing a to-do from being missed once the discussion is resolved. 😄 

@rarkins said this in a discussion (#10316):

> We should make it that way. Regex manager defaults to semver versioning so then the datasource default does not take effect. We should remove the regex manager versioning default.

Probably a good idea to read the full discussion. 😉 

**Did you already have any implementation ideas?**

Remove regex manager versioning default.

### PR DESCRIPTION
**Pull request by secustor**
## Changes:

This PR removes the defaulting to semver if no other versioning is defined. 

## Context:

Closes #10320

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by viceice — APPROVED** [review]
If i remember correctly renovate will fallback to semver in code anyway when versioning is undefined. Maybe we need to check that no warning is logged.

**Comment by viceice** [comment]
https://github.com/renovatebot/renovate/blob/74d7691befa635c1220d8569936273bfb950e748/lib/versioning/index.ts#L18-L21

@rarkins Are we ok with this debug message then? I think that can happen with this PR.

**Comment by rarkins** [comment]
Maybe lower it to trace and say "Defaulting to semver versioning"?

**Review by viceice — CHANGES_REQUESTED** [review]
@secustor Please change the logging as mentioned.

**Review by rarkins — DISMISSED** [review]

File: lib/versioning/index.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ export const getVersionings = (): Map<
 
 export function get(versioning: string): VersioningApi {
   if (!versioning) {
-    logger.debug('Missing versioning');
+    logger.trace('Missing versioning');
```
Comment (viceice) [inline lib/versioning/index.ts:19]:
```suggestion
    logger.trace('Missing versioning, using semver as fallback.');
```

**Review by viceice — COMMENTED** [review]
I think this makes the message more clear.

**Review by viceice — APPROVED** [review]

**Pull request merged**